packages/web-api-client/src/a11y-service-credential.ts

```diff
@@ -30,14 +59,16 @@ export class A11yServiceCredential {
         });
     }
 
-    public async signRequest(request: typeof requestPromise): Promise<typeof requestPromise> {
-        const accessToken = await this.getToken();
-        const authOptions: requestPromise.RequestPromiseOptions = {
-            headers: {
-                authorization: `${accessToken.tokenType} ${accessToken.accessToken}`,
-            },
-        };
+    private async handleGetTokenError(err: Error): Promise<void> {
+        this.logger.logError(`Auth getToken call failed with error: ${JSON.stringify(err)}`);
 
-        return request.defaults(authOptions);
+        if (this.sleepBetweenRetries) {
+            await this.sleep(1000);
```

Comment:
this code should be in Retry helper.
RetryHelper.retry<T>(callback: ()=> T, maxAttempts, retryIntervalInMilliSec): Promise<T>